internal/promote/pull_requests.go

```diff
@@ -0,0 +1,176 @@
+		return "", fmt.Errorf("unexpected status code return while opening a pull request: %d", response.StatusCode)
+	}
+
+	var data createPullRequestResponse
+	defer response.Body.Close()
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return "", errors.Wrap(err, "can't read response body")
+	}
+
+	if err := json.Unmarshal(body, &data); err != nil {
+		return "", errors.Wrap(err, "unmarshalling response failed")
+	}
+
+	err = updatePullRequestAssignee(data.Number, user)
+	if err != nil {
+		return "", errors.Wrapf(err, "updating assignees failed (pullRequestID: %d, user: %s)", data.Number, user)
+	}
+	return data.HTMLURL, nil
+}
+
+func updatePullRequestAssignee(pullRequestID int, user string) error {
+	authToken, err := github.AuthToken()
+	if err != nil {
+		return errors.Wrap(err, "reading auth token failed")
+	}
+
+	requestBody, err := buildPullRequestAssigneesRequestBody(user)
+	if err != nil {
+		return errors.Wrap(err, "building assignees request body failed")
+	}
+
+	request, err := http.NewRequest("PATCH", fmt.Sprintf("https://api.github.com/repos/elastic/package-storage/issues/%d", pullRequestID),
+		bytes.NewReader(requestBody))
+	if err != nil {
+		return errors.Wrap(err, "creating new HTTP request failed")
+	}
+
+	request.Header.Add("Authorization", fmt.Sprintf("token %s", authToken))
+	request.Header.Add("Content-Type", "application/json")
+	response, err := http.DefaultClient.Do(request)
+	if err != nil {
+		return errors.Wrap(err, "making HTTP call failed")
+	}
+
+	if response.StatusCode < 200 || response.StatusCode > 299 {
+		return fmt.Errorf("unexpected status code return while opening a pull request: %d", response.StatusCode)
+	}
+
+	var data createPullRequestResponse
+	defer response.Body.Close()
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return errors.Wrap(err, "can't read response body")
+	}
+
+	if err := json.Unmarshal(body, &data); err != nil {
+		return errors.Wrap(err, "unmarshalling response failed")
+	}
+
+	if len(data.Assignees) == 0 {
+		return errors.New("no users assigned to the pull request")
+	}
+	return nil
+}
+
+func buildPullRequestRemoveDescription(sourceStage string, revisions PackageRevisions) string {
+	var builder strings.Builder
+	builder.WriteString(fmt.Sprintf("This PR removes packages from `%s`.\n", sourceStage))
+	builder.WriteString("\n")
+	builder.WriteString("Removed packages:\n")
+	for _, revision := range revisions {
+		builder.WriteString(fmt.Sprintf("* `%s-%s`\n", revision.Name, revision.Version))
+	}
+	return builder.String()
+}
+
+func buildPullRequestPromoteDescription(sourceStage, destinationStage string, revisions PackageRevisions) string {
+	var builder strings.Builder
+	builder.WriteString(fmt.Sprintf("This PR promotes packages from `%s` to `%s`.\n", sourceStage, destinationStage))
+	builder.WriteString("\n")
+	builder.WriteString("Promoted packages:\n")
+	for _, revision := range revisions {
```

Comment:
Nit: I think you could loop over `revisions.Strings()` here. Then you only need to prefix the string with `* ` inside the loop.

Reply:
Fixed